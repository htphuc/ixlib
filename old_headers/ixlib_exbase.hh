#include <ixlib_exbase.hpp>
