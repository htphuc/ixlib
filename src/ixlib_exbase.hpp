// ----------------------------------------------------------------------------
//  Description      : Exception handling
// ----------------------------------------------------------------------------
//  (c) Copyright 1996 by iXiONmedia, all rights reserved.
// ----------------------------------------------------------------------------




#ifndef IXLIB_EXBASE
#define IXLIB_EXBASE




#include <stdexcept>
#include <ixlib_base.hpp>




// constants ------------------------------------------------------------------
#define EX_INFOMAX 255




// throw macro ----------------------------------------------------------------
#define EX_THROW(TYPE,CODE)\
  throw ::ixion::TYPE##_exception(CODE,NULL,__FILE__,__LINE__);
#define EX_THROWINFO(TYPE,CODE,INFO)\
  throw ::ixion::TYPE##_exception(CODE,(char const *) INFO,__FILE__,__LINE__);
#define EX_CATCHCODE(TYPE,CODE,HANDLER)\
  catch (TYPE##_exception &ex) { \
    if (ex.Error != CODE) throw; \
    HANDLER \
  }
#define EX_CONVERT(TYPE,CODE,DESTTYPE,DESTCODE)\
  catch (TYPE##_exception &ex) { \
    if (ex.Error != CODE) throw; \
    throw DESTTYPE##_exception(DESTCODE,ex.Info,__FILE__,__LINE__); \
  }
  
  
  
  
// xBaseException -------------------------------------------------------------
namespace ixion {
  typedef unsigned int TErrorCode;
  
  
  
  
  struct base_exception : public std::exception {
    TErrorCode          Error;
    char                *Module;
    TIndex             	Line;
    char                *Category;
    bool		HasInfo;
    char 		Info[EX_INFOMAX+1];
    static char		RenderBuffer[EX_INFOMAX+1+100];
  
    base_exception(TErrorCode error,char const *info = NULL,char *module = NULL,
      TIndex line = 0,char *category = NULL);
    char const *what() const throw ();
    virtual const char *getText() const = 0;
    };
  }



#endif
