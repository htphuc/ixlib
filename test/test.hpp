// ----------------------------------------------------------------------------
//  Description      : Cheap testing framework
// ----------------------------------------------------------------------------
//  (c) Copyright 1998 by iXiONmedia, all rights reserved.
// ----------------------------------------------------------------------------




#ifndef IXLIB_TEST
#define IXLIB_TEST




#include <iostream>
#include <ixlib_exbase.hpp>




using namespace std;




#define BEGIN_TEST \
int main() {\
  try { \
    TSize errcount = 0,testcount = 0; 

#define END_TEST \
  if (errcount) \
    cout << errcount << " out of " << testcount \
      << " test(s) failed." << endl; \
  else \
    cout << "all " << testcount << " tests passed." << endl; \
  return errcount; \
  } \
  catch (exception &e) { \
    cerr << "*** exception exit. bad." << endl; \
    cerr << "*** " << e.what() << endl; \
    return 1; \
    } \
  catch (...) { \
    cerr << "*** exception exit. bad." << endl; \
    return 1; \
    } \
}

#define PASSED { testcount++; }

#define FAILED { testcount++; errcount++; \
  cerr << "*** Test " << testcount << " failed." << endl; }

#define FAILEDEX { testcount++; errcount++; \
  cerr << "*** Test " << testcount << " failed due to an exception." << endl; }
  
#define TEST(INVARIANT) \
  try { \
    if (INVARIANT) PASSED else { \
      FAILED \
      cerr << "*** " << #INVARIANT << endl; \
      }\
    } \
  catch (exception &e) { \
    FAILEDEX \
    cerr << "*** " << e.what() << endl; \
    }




#endif
