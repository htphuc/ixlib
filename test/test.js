function TEST(NAME,CONDITION) {
  if (!(CONDITION)) { 
    failed_tests += NAME + " ";
    }
  tests++;
  }

var failed_tests = "",tests = 0;
// test null ------------------------------------------------------------------
{ var counter;
  TEST("null 1",null == null);
  TEST("null 2",counter == null);
  TEST("null 3",null == counter);
  TEST("null 4",3 != null);
  TEST("null 5",!(3 == null));
  TEST("null 6",4.0 != null);
  TEST("null 7",!(4.0 == null));
  TEST("null 8","stuff" != null);
  TEST("null 9",!("stuff" == null));
  }

// test strings ---------------------------------------------------------------
{ TEST("string 1","0123456".charAt(3) == "3");
  TEST("string 2","0123456".charCodeAt(3) == 51);
  TEST("string 3","abc".concat("123","456") == "abc123456");
  TEST("string 4","gagagaratzgaratzga".indexOf("ratz") == 6);
  TEST("string 5","gagagaratzgaratzga".lastIndexOf("ratz") == 12);
  TEST("string 6","gagagaratzgaratzga".slice(6,10) == "ratz");
  TEST("string 7","gagagaratzgaratzga".substring(6,10) == "ratz");
  var a = "de en fr ch hallo";
  TEST("string 8",a.split(" ").length == 5);
  TEST("string 9",a.split("").length == a.length);
  TEST("string 10",a.split().length == a.length);
  TEST("string 11",a.split(" ",3).length == 3);
  TEST("string 12",a.split(" ").join(" ") == "de en fr ch hallo");
  }

// test prefix stuff ----------------------------------------------------------
{ var counter = 5;

  TEST("prefix 1",counter == 5);
  TEST("prefix 2",counter++ == 5);
  (++counter)++;
  TEST("prefix 3",counter == 8);
  }

// test variable declaration --------------------------------------------------
{ var broesel = 5,bramm = 1+broesel;
  const brusel = 1;

  TEST("variables 1",broesel == 5);
  TEST("variables 2",bramm == 6);
  TEST("variables 3",brusel == 1);
  
  var u = "HALLO";
  u += "DU!";
  TEST("variables 4",u == "HALLODU!");
  }

// test if --------------------------------------------------------------------
{ if (0) {
    TEST(0,"if 1");
    }
  else {
    }
  
  if (1) {
    }
  else {
    TEST(0,"if 2");
    }
  }


// test while -----------------------------------------------------------------
{ var counter = 0;
  while (counter < 10) counter++;
  TEST("while 1",counter == 10);
  }


// test do-while --------------------------------------------------------------
{ var counter = 0;
  do {
    counter++;
  } while (counter < 10)
  TEST("do-while 1",counter == 10);
  }

// test for -------------------------------------------------------------------
{ var i = 0;

  // this must not cause an error , because for has its own scope
  var counter; 
  
  for (var counter = 0;counter < 10;counter++) {
    i++;
    }
  TEST("for 1",i == 10);
  }

// test for-in ----------------------------------------------------------------
{ const a = new Array(1,1,2,6,24,120,720,5040,40320,362880,3628800);
  var sum = 0;
  
  for (var i in a)
    sum += i;
  TEST("for-in 1",sum == 4037914);
  }

// test scoping ---------------------------------------------------------------
{ var something = 5;
  something;
  
  { something++;
    var something = 7;
    something++;
    }
  TEST("scoping 1",something == 6);
  }

// test functions -------------------------------------------------------------
{ function faculty(n) {
    var counter,result = 1;
    for (counter = 1;counter <= n ;counter++)
      result *= counter;
    return result;
    }
  
  function faculty_recursive(n) {
    if (n <= 1) return 1;
    else return faculty_recursive(n-1)*n;
    }
  
  function product(a,b,c,d) {
    return a*b*c*d;
    }
  
  function spend_some_time() {
    var i = 0;
    while (i < 1000) i++;
    return;
    }

  function assign(x) {
    x = 17;
    }
  
  TEST("functions 1",product(1,2,3,4) == 24);
  TEST("functions 2",faculty(10) == 3628800);
  TEST("functions 3",faculty_recursive(10) == 3628800);
  TEST("functions 4",faculty_recursive(90) == faculty(90));
  spend_some_time();
  
  var u = 4;
  assign(u);
  TEST("functions 5",u == 4);
  }

// test empty statement -------------------------------------------------------
{ function faculty(n) {
    var counter,result = 1;
    for (counter = 1;counter <= n ;counter++)
      result *= counter;
    return result;
    }
  
  var a = new Array(1,1,2,6,24,120,720,5040,40320,362880,3628800),u = 0;
  
  for (var i in a)
    TEST("for-in 1",i == faculty(u++));
  }

// test non-local exits -------------------------------------------------------
{ var i;
  for (i = 0;i < 2000;i++) {
    if (i == 100) break;
    }
  TEST("exits 1",i == 100);
  
  var counter = 0;
  
  for (i = 0;i < 200;i++) {
    if (i % 2 == 0) continue;
    counter++;
    }
  TEST("exits 2",counter == 100);
  
  counter = 0;
  outer:
  for (i = 0;i < 20;i++) {
    for (var j = 0;j < 20;j++) {
      if (j == i) continue outer;
      counter++;
      }
    }
  TEST("exits 3",counter == 190);
  counter;
  
  var flag = 0;
  
  non_local_exit: {
    var i = 0;
    i++;
    flag = 2;
    break non_local_exit;
    flag = 1;
    }
  TEST("exits 4",flag == 2);
  
  i = 0;
  counter = 0;
  loop: while (i < 10) {
    another_loop: while (i < 20) {
      break loop;
      counter++;
      }
    counter++;
    }
  TEST("exits 5",counter == 0);
  }

// test switch statement ------------------------------------------------------
{ var count1 = 0,count2 = 0,count3 = 0,defcount = 0,i;

  for (i = 0;i < 10;i++) {
    switch (i) {
      0;
      case 1: count1++;
      case 2: count2++;
      case 3: count3++; break;
      default: defcount++;
      }
    }
  TEST("switch",count1 == 1 && count2 == 2 && count3 == 3 && defcount == 7);
  }

// test arrays ----------------------------------------------------------------
{ function faculty(n) {
    var counter,result = 1;
    for (counter = 1;counter <= n ;counter++)
      result *= counter;
    return result;
    }
  
  var a = new Array(10);
  a[5] = 4;
  TEST("arrays 1",a[5] == 4);
  
  var i;
  a = new Array(1,1,2,6,24,120,720,5040,40320,362880,3628800);
  for (i = 0;i <= 10;i++)
    TEST("arrays "+(2+i),a[i] == faculty(i));
  
  TEST("arrays 13",a.length == 11);
  a.reverse();
  TEST("arrays 14",a[0] == 3628800);
  a.reverse();
  for (i = 0;i <= 10;i++)
    TEST("arrays "+(2+i),a[i] == faculty(i));
  TEST("arrays 15",a.pop() == 3628800);
  TEST("arrays 16",a.length == 10);
  a.push(15);
  TEST("arrays 17",a[10] == 15);
  
  for (i = 11;i <= 2000;i++)
    TEST("arrays 18",a[i] == null);
  }

// test library ---------------------------------------------------------------
{ var u = 20;
  TEST("lib 1",eval("4+3+2+1;") == 10);
  TEST("lib 2",parseInt("0xaffe") == 0xAFFE);
  TEST("lib 3",parseFloat("3.14") == 3.14);
  TEST("lib 10",0 <= Math.random() && Math.random() <= 1);
  TEST("lib 11",Math.abs(Math.sin(Math.PI)) < 0.0001);
  
  var usin = Math.sin;
  TEST("lib 12",Math.abs(usin(Math.PI)) < 0.0001);
  }

// test classes ---------------------------------------------------------------
{ 
  class flupp {
    static const gravity = 9.81;
    static var u = new Array();
    var v = new Array();
    
    constructor function flupp(x) {
      v.push(x);
      }
    static function getGravity() {
      return gravity;
      }
    }
  class gummiflupp extends flupp {
    var elasticity;
    
    constructor function gummiflupp(_elasticity) {
      elasticity = _elasticity;
      super(17);
      }
    function f(x) {
      return elasticity*Math.cos(x)*getGravity();
      }
    }
  
  TEST("class 1",gummiflupp.gravity == gummiflupp.getGravity());
  
  var f = new gummiflupp(7);
  TEST("class 2",f.f(0) > 7*9.8);
  TEST("class 3",f.v[0] == 17);

  var u = 3;
  class scope_test {
    var u = 4;
    function test() {
      u;
      }
    };
  var v = new scope_test();
  TEST("class 4",v.test() == 4);
  }

// output main result ---------------------------------------------------------
if (failed_tests == "") {
  "OK: " + tests.toString() + " tests completed";
  }
else {
  "failed: " + failed_tests;
  }
