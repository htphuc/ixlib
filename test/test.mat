3x3
9 -7 2
-7 11 -3
2 -3 4
4x4
4 0 -2 -2
0  1 3  0
-2 3 11 3
-2 0 3  14
4x4
4 0 -2 -2
0  1 3  0
-2 3 11 3
-2 0 3  14
