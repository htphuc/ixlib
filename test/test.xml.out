<!-- Generated by ixlib XML -->
<animation >
  <font baseline="16" bottomline="16" name="New Font" topline="0" >
    <glyph base_x="0" base_y="16" bound_h="1" bound_w="1" bound_x="0" bound_y="0" code="32" width="16" />
    <glyph base_x="-7" base_y="16" bound_h="16" bound_w="4" bound_x="1" bound_y="0" code="33" width="16" />
    <glyph base_x="-4" base_y="16" bound_h="4" bound_w="10" bound_x="5" bound_y="0" code="34" width="16" />
    <glyph base_x="0" base_y="16" bound_h="6" bound_w="6" bound_x="15" bound_y="0" code="39" width="16" />
    <glyph base_x="-3" base_y="16" bound_h="16" bound_w="10" bound_x="21" bound_y="0" code="40" width="16" />
    <glyph base_x="-4" base_y="16" bound_h="16" bound_w="11" bound_x="31" bound_y="0" code="41" width="16" />
    <glyph base_x="-5" base_y="12" bound_h="8" bound_w="8" bound_x="42" bound_y="0" code="43" width="16" />
    <glyph base_x="0" base_y="6" bound_h="6" bound_w="6" bound_x="50" bound_y="0" code="44" width="16" />
    <glyph base_x="-4" base_y="10" bound_h="4" bound_w="9" bound_x="56" bound_y="0" code="45" width="16" />
    <glyph base_x="-7" base_y="5" bound_h="4" bound_w="3" bound_x="65" bound_y="0" code="46" width="16" />
    <glyph base_x="-1" base_y="16" bound_h="16" bound_w="15" bound_x="68" bound_y="0" code="48" width="16" />
    <glyph base_x="-2" base_y="16" bound_h="16" bound_w="9" bound_x="83" bound_y="0" code="49" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="92" bound_y="0" code="50" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="107" bound_y="0" code="51" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="122" bound_y="0" code="52" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="137" bound_y="0" code="53" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="152" bound_y="0" code="54" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="167" bound_y="0" code="55" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="182" bound_y="0" code="56" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="197" bound_y="0" code="57" width="16" />
    <glyph base_x="-6" base_y="13" bound_h="10" bound_w="4" bound_x="212" bound_y="0" code="58" width="16" />
    <glyph base_x="-2" base_y="12" bound_h="12" bound_w="7" bound_x="216" bound_y="0" code="59" width="16" />
    <glyph base_x="-4" base_y="11" bound_h="9" bound_w="9" bound_x="223" bound_y="0" code="61" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="14" bound_x="232" bound_y="0" code="63" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="246" bound_y="0" code="65" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="261" bound_y="0" code="66" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="276" bound_y="0" code="67" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="291" bound_y="0" code="68" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="306" bound_y="0" code="69" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="321" bound_y="0" code="70" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="336" bound_y="0" code="71" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="351" bound_y="0" code="72" width="16" />
    <glyph base_x="-5" base_y="16" bound_h="16" bound_w="7" bound_x="366" bound_y="0" code="73" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="373" bound_y="0" code="74" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="388" bound_y="0" code="75" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="403" bound_y="0" code="76" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="418" bound_y="0" code="77" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="433" bound_y="0" code="78" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="448" bound_y="0" code="79" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="463" bound_y="0" code="80" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="478" bound_y="0" code="81" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="493" bound_y="0" code="82" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="508" bound_y="0" code="83" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="523" bound_y="0" code="84" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="538" bound_y="0" code="85" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="553" bound_y="0" code="86" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="568" bound_y="0" code="87" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="583" bound_y="0" code="88" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="598" bound_y="0" code="89" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="613" bound_y="0" code="90" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="628" bound_y="0" code="97" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="643" bound_y="0" code="98" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="658" bound_y="0" code="99" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="673" bound_y="0" code="100" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="688" bound_y="0" code="101" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="703" bound_y="0" code="102" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="718" bound_y="0" code="103" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="733" bound_y="0" code="104" width="16" />
    <glyph base_x="-5" base_y="16" bound_h="16" bound_w="7" bound_x="748" bound_y="0" code="105" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="755" bound_y="0" code="106" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="770" bound_y="0" code="107" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="785" bound_y="0" code="108" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="800" bound_y="0" code="109" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="815" bound_y="0" code="110" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="830" bound_y="0" code="111" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="845" bound_y="0" code="112" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="860" bound_y="0" code="113" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="875" bound_y="0" code="114" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="890" bound_y="0" code="115" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="905" bound_y="0" code="116" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="920" bound_y="0" code="117" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="935" bound_y="0" code="118" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="950" bound_y="0" code="119" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="965" bound_y="0" code="120" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="980" bound_y="0" code="121" width="16" />
    <glyph base_x="0" base_y="16" bound_h="16" bound_w="15" bound_x="995" bound_y="0" code="122" width="16" />
  </font>
  <images >
    <image name="bazooka-d05" src="mole_bazooka_look_05.png" />
    <image name="bazooka-d03" src="mole_bazooka_look_03.png" />
    <image name="bazooka-d01" src="mole_bazooka_look_01.png" />
    <image name="bazooka-d15" scale_x="-1" src="mole_bazooka_look_05.png" />
    <image name="bazooka-d17" scale_x="-1" src="mole_bazooka_look_03.png" />
    <image name="bazooka-d19" scale_x="-1" src="mole_bazooka_look_01.png" />
    <image name="bazooka-d05-walk00" src="mole_bazooka_walk00_05.png" />
    <image name="bazooka-d15-walk00" scale_x="-1" src="mole_bazooka_walk00_05.png" />
    <image name="bazooka-d05-crouch-1" scale_y="0.9" src="mole_bazooka_look_05.png" />
    <image name="bazooka-d05-crouch-2" scale_y="0.85" src="mole_bazooka_look_05.png" />
    <image name="bazooka-d15-crouch-1" scale_x="-1" scale_y="0.9" src="mole_bazooka_look_05.png" />
    <image name="bazooka-d15-crouch-2" scale_x="-1" scale_y="0.85" src="mole_bazooka_look_05.png" />
  </images>
  <regexes >
    <regex original="d0[0-2]" replacement="d01" />
    <regex original="d0[3-4]" replacement="d03" />
    <regex original="d0[5-9]" replacement="d05" />
    <regex original="d1[1-5]" replacement="d15" />
    <regex original="d1[6-7]" replacement="d17" />
    <regex original="d1[7-9]" replacement="d19" />
    <regex original="d0.-prejump" replacement="d05-crouch" />
    <regex original="d1.-prejump" replacement="d15-crouch" />
    <regex original="d0.-jump" replacement="d01" />
    <regex original="d1.-jump" replacement="d19" />
    <regex original="d0.-land" replacement="d05-crouch" />
    <regex original="d1.-land" replacement="d15-crouch" />
    <regex original="d0.-walk" replacement="d05-walk" />
    <regex original="d1.-walk" replacement="d15-walk" />
  </regexes>
  <script >
    <on parameter="direction" will_return >
      <return kill_delay />
    </on>
    <on new="land" parameter="state" will_return >
      <priority push />
      <priority value="5" />
      <frame name="$(weapon)-d$(direction)-land-1" />
      <delay time="30" />
      <frame name="$(weapon)-d$(direction)-land-2" />
      <delay time="30" />
      <frame name="$(weapon)-d$(direction)-land-1" />
      <delay time="30" />
      <priority pop />
      <return kill_delay />
    </on>
    <on new="prejump" parameter="state" priority="10" will_return >
      <frame name="$(weapon)-d$(direction)-prejump-1" />
      <delay time="30" />
      <label name="prejump_loop" />
      <frame name="$(weapon)-d$(direction)-prejump-2" />
      <delay time="10000" />
      <goto target="prejump_loop" />
      <stop />
    </on>
    <on new="jump" parameter="state" priority="10" >
      <frame name="$(weapon)-d$(direction)-prejump-1" />
      <delay time="30" />
      <signal name="jump" />
      <priority push />
      <priority value="5" />
      <frame name="$(weapon)-d$(direction)-jump" />
      <delay time="400" />
      <priority pop />
      <return kill_delay />
    </on>
    <on new="walk" parameter="state" >
      <label name="walk_loop" />
      <frame name="$(weapon)-d$(direction)" />
      <delay time="150" />
      <frame name="$(weapon)-d$(direction)-walk00" />
      <delay time="150" />
      <goto target="walk_loop" />
    </on>
    <on new="stand" parameter="state" >
      <goto target="main_loop" />
    </on>
    <on new="fly" parameter="state" >
      <goto target="main_loop" />
    </on>
    <label name="main_loop" />
    <frame name="$(weapon)-d$(direction)" />
    <delay time="10000" />
    <goto target="main_loop" />
  </script>
</animation>
